packages/tracing-processing/src/services/enricherService.ts

```diff
@@ -70,36 +69,33 @@ export function dbServiceBuilder(db: DB) {
               );
 
               if (!consumer) {
-                return {
-                  purposeId: record.purpose_id,
-                  status: record.status,
-                  errorCode: PurposeErrorCodes.CONSUMER_NOT_FOUND,
-                  message: PurposeErrorCodes.CONSUMER_NOT_FOUND,
-                  rowNumber: record.rowNumber,
-                };
+                throw getEnrichedPurposeError(
+                  `Consumer ${tracing.tenantId} not found for tracingId: ${tracing.tracingId}, purpose_id: ${record.purpose_id}`,
+                );
               }
 
               if (!eService) {
-                return {
-                  purposeId: record.purpose_id,
-                  status: record.status,
-                  errorCode: PurposeErrorCodes.ESERVICE_NOT_FOUND,
-                  message: PurposeErrorCodes.ESERVICE_NOT_FOUND,
-                  rowNumber: record.rowNumber,
-                };
+                throw getEnrichedPurposeError(
+                  `Eservice ${fullPurpose.eservice_id} not found for tracingId: ${tracing.tracingId}, purpose_id: ${record.purpose_id}`,
+                );
               }
 
               const tenantEservice = await db.oneOrNone<EserviceSchema>(
-                `SELECT * FROM tracing.eservices WHERE (producer_id = $1 OR consumer_id = $1)`,
+                `SELECT * FROM tracing.eservices WHERE producer_id = $1`,
```

Comment:
this can return more than one result, as an organization may have multiple eservices.
You probably need to filter also for eservice_id. At this point can we use [this](https://github.com/pagopa/interop-tracing-core/pull/9/files/d0008b8e90281c526a5a927157f1023ccf06bfe7..51fb8afaa47a080f805b3f467fdca62f96f842cb#diff-7d6e39674e943de6a8bcf94e7d87f1436c8414239e4df9665bc7e6316abb4bf3R58)?